conans/client/remote_manager.py

```diff
@@ -161,10 +204,9 @@ def addfile(name, file_info, tar):
 def uncompress_files(files, folder, name):
     try:
         for file_name, content in files:
-            if os.path.basename(file_name) != name:
-                save(os.path.join(folder, file_name), content)
-            else:
-                #  Unzip the file
+            save(os.path.join(folder, file_name), content)
```

Comment:
Some numbers:

With 0.12 => Compressing Boost tgz ~1GB Ram
                     Uploading Boost package ~300MB Ram 
     
With this PR => Memory always keeps low ~100MB
                   
Uploads => About 30% faster (due not managing big contents dicts) (without counting the re-tgz again)